landingpage/src/components/TransactionTool.tsx

```diff
@@ -0,0 +1,175 @@
+import MarkedTextWithCopy from './MarkedTextWithCopy'
+import { useEffect, useState } from 'react'
+import DropDownContent from './HeaderWithSearch/DropDownContent'
+import AlterConfigForName, { AlterOptionKey } from './AlterConfigForName'
+
+interface TransactionToolProps {
+	mode: 'claimAndInscribe'|'inscribe'
+	name: string;
+	history?: PlebNameHistory;
+}
+
+type InscriptionMap = Partial<Record<AlterOptionKey, string>>;
+
+export const TransactionTool: React.FC<TransactionToolProps> = ({ name, mode, history }) => {
+	const [senderAddress, setSenderAddress] = useState(history?.getData().owner ??'')
+	const [validSenderAddress, setValidSenderAddress] = useState<string|undefined>(undefined)
+	const [senderUtxo, setSenderUtxo] = useState<bitcoinExplorer.UTXO|undefined>(undefined)
+	const [senderUtxoStatus, setSenderUtxoStatus] = useState<'addressNeeded'|'fetching'|'ok'|Error>('addressNeeded')
+	const [inscriptionMap, setInscriptionMap] = useState<InscriptionMap>({})
+	const [transaction, setTransaction] = useState<{
+		transaction: {toHex: () => string}
+		senderAddressError?: Error
+		senderUtxoError?: Error
+	} | undefined>(undefined)
+
+	useEffect(() => {
+		const tx = generateTransaction({name, senderAddress, senderUtxo, inscriptions: Object.values(inscriptionMap), mode})
+		setTransaction(tx)
+		setValidSenderAddress(tx && !tx.senderAddressError ? senderAddress : undefined)
+	}, [name, senderAddress, senderUtxo, inscriptionMap, mode])
+
+	useEffect(() => {
+		if (!validSenderAddress) {
+			setSenderUtxoStatus('addressNeeded')
+			return
+		}
+		setSenderUtxoStatus('fetching')
+		bitcoinExplorer.explorerAdapter.getUtxosOfAddress(validSenderAddress).then(utxos => {
+			if (utxos.length < 1) {
+				setSenderUtxoStatus(new Error(`Address does not have unspent transaction outputs (UTXOs), send something to '${validSenderAddress}' first.`))
+				setSenderUtxo(undefined)
+			} else {
+				setSenderUtxoStatus('ok')
+				setSenderUtxo(utxos[0])
+			}
+		}).catch(error => {
+			setSenderUtxoStatus(toError(error))
+			setSenderUtxo(undefined)
+		})
+	}, [validSenderAddress])
+
+	const validTransaction: boolean = !transaction?.senderAddressError && !transaction?.senderUtxoError && senderUtxoStatus === 'ok'
+	const historyData = history?.getData();
+
+	return (
+
+		<div className='flex flex-col '>
+			{!history && 	
+			<div className="modifyConfigSelect flex flex-row flex-wrap items-center justify-start gap-3">	<label>
+				Your Address:{' '}
+				</label>
+				<input
+					placeholder='bc1q88758c9etpsvntxncg68ydvhrzh728802aaq7w'
+					value={senderAddress}
+					onChange={(event: React.ChangeEvent<HTMLInputElement>) => setSenderAddress(event.target.value)}
+					className="border-gray-30 flex-1 rounded-md border bg-gray-100 px-3 py-2 text-blue-950 placeholder:text-gray-500"	
+				/>
+			<br /></div>}
+		
+
+				{historyData && Object.entries(historyData).map((entry, index) => (	
```

Comment:
find it more intuitive if in the form only things are displayed that are encoded in the transaction. So only entries that are added (or overwritten). Then the form is also shorter.

Reply:
Yes. This is just a workaround until we have a "+" option. We can remove it for now.